### ISSUE TITLE
H.name and H['name'] are inconsistent

### ISSUE THREAD

**Issue by leotrs** [ComplexGroupInteractions/xgi#42]
After #41, the code `H['name']` will access hypergraph attributes and therefore should (hopefully) always return the same as `H.name`. However, if the name is not set, we will get, after that PR is merged, the following:
```python
H.name
# -> ""

H['name']
# -> KeyError: 'name'
```

**Comment by nwlandry** [comment]
Good point. What about in the `__getitem__` property, we have:
```
return self._hypergraph.get(attr, "")
```

**Comment by leotrs** [comment]
I would be against this solution because having a key whose value is the empty string is different than not having the key at all.

**Comment by nwlandry** [comment]
Should be fixed with #41.
